### PR TITLE
Fix mini typos in docs

### PR DESCRIPTION
**Pull request by svilupp**
As part of the JuliaCon hackathon, I have run all the docs through Grammarly.

No changes were in the executable code (ie, only comments, labels, docs)

There are several typos left in the other packages (eg, variations of "probablistic") but this is everything I could find here.

I'm happy to withdraw this PR if it's not helpful.

### CODE REVIEW

File: docs/src/about_mlj.md [hunk 1]
```diff
@@ -156,8 +156,8 @@ Extract:
 
 * Extensible [tuning
   interface](https://github.com/JuliaAI/MLJTuning.jl),
-  to support growing number of optimization strategies, and designed
-  to play well with model composition.
+  to support the growing number of optimization strategies, designed
+  to play well with the model composition.
```
Comment (ablaom) [inline docs/src/about_mlj.md:160]:
```suggestion
  to support a growing number of optimization strategies, and designed
  to play well with model composition.
```

File: docs/src/adding_models_for_general_use.md [hunk 2]
```diff
@@ -60,7 +60,7 @@ following packages is also helpful:
   algorithm needs input data in a novel format).
 
 In MLJ, the basic interface exposed to the user, built atop the model
-interface described here, is the *machine interface*. After a first
+interface described here, is the *machine interface*. After the first
```
Comment (ablaom) [inline docs/src/adding_models_for_general_use.md:63]:
```suggestion
interface described here, is the *machine interface*. After a first
```

File: docs/src/adding_models_for_general_use.md [hunk 3]
```diff
@@ -539,14 +539,14 @@ Note that while `Xnew` generally consists of multiple observations
...
-probablility distribution `p(X |y)` above).
+probability distribution `p(X |y)` above).
 
 
 #### Prediction types for deterministic responses.
 
 In the case of `Deterministic` models, `yhat` should have the same
 scitype as the `y` passed to `fit` (see above). If `y` is a
-`CategoricalVector` (classification) then elements of the predition
+`CategoricalVector` (classification) than elements of the prediction
```
Comment (ablaom) [inline docs/src/adding_models_for_general_use.md:549]:
```suggestion
`CategoricalVector` (classification) then elements of the prediction
```

File: docs/src/adding_models_for_general_use.md [hunk 4]
```diff
@@ -610,7 +610,7 @@ distribution per row of `Xnew`.
 A *distribution* is some object that, at the least, implements
 `Base.rng` (i.e., is something that can be sampled).  Currently, all
 performance measures (metrics) defined in MLJBase.jl additionally
-assume that a distribution is either:
+assume that distribution is either:
```
Comment (ablaom) [inline docs/src/adding_models_for_general_use.md:613]:
```suggestion
assume that a distribution is either:
```

File: docs/src/adding_models_for_general_use.md [hunk 5]
```diff
@@ -685,7 +685,7 @@ instead of vectors. See
 
 See
 [LinearBinaryClassifier](https://github.com/JuliaAI/MLJModels.jl/blob/master/src/GLM.jl)
-for an example of a Probabilistic classifier implementation.
+for example of a Probabilistic classifier implementation.
```
Comment (ablaom) [inline docs/src/adding_models_for_general_use.md:688]:
```suggestion
for an example of a Probabilistic classifier implementation.
```

File: docs/src/adding_models_for_general_use.md [hunk 6]
```diff
@@ -800,7 +800,7 @@ target_scitype(SomeSupervisedModel) = Table(Continuous, Finite{2})
 restricts to tables with continuous or binary (ordered or unordered)
 columns.
 
-For predicting variable length sequences of, say, binary values
+For predicting variable-length sequences of, say, binary values
```
Comment (ablaom) [inline docs/src/adding_models_for_general_use.md:803]: Variable is an adjective (not a noun acting as an adjective) so I would beg to differ with Grammarly here. 

```suggestion
For predicting variable length sequences of, say, binary values
```

File: docs/src/composing_models.md [hunk 7]
```diff
@@ -443,7 +443,7 @@ This completes the export process.
 Notes:
 
 - The line marked `###`, where the new exported model's hyperparameter
-  `regressor` is spliced into the network, is the only modification to
+  `regressor` is spliced into the network, and is the only modification to
```
Comment (ablaom) [inline docs/src/composing_models.md:446]:
```suggestion
  `regressor` is spliced into the network, is the only modification to
```

File: docs/src/composing_models.md [hunk 8]
```diff
@@ -600,7 +600,7 @@ zhat = node((y1, y2)->sqrt.(y1.*y2), predict(mach1, W), predict(mach2, W))
 
 ### More `node` examples
 
-Here are some examples taken from MLJ source
+Here are some examples taken from the MLJ source
```
Comment (ablaom) [inline docs/src/composing_models.md:603]:
```suggestion
Here are some examples taken from the MLJ source code
```

File: docs/src/controlling_iterative_models.md [hunk 9]
```diff
@@ -9,7 +9,7 @@ such as a learning rate, in response to the behavior of these
 estimates.
 
 Some iterative model implementations enable some form of automated
-control, with the method and options for doing so varying from model
+control, with the method and options for doing so vary from model
```
Comment (ablaom) [inline docs/src/controlling_iterative_models.md:12]:
```suggestion
control, with the method and options for doing so varying from model
```

File: docs/src/controlling_iterative_models.md [hunk 10]
```diff
@@ -118,7 +118,7 @@ control                                                        | description
 [`WithMachineDo`](@ref MLJIteration.WithMachineDo)`(f->mach->@info("report: $mach))`| Call `f(mach)` wher `mach` is the training machine in its current state    | yes
 [`Save`](@ref MLJIteration.Save)`(filename="machine.jls")`|Save current training machine to `machine1.jls`, `machine2.jsl`, etc                         | yes
 
-> Table 1. Atomic controls. Some advanced options omitted.
+> Table 1. Atomic controls. Some advanced options were omitted.
```
Comment (ablaom) [inline docs/src/controlling_iterative_models.md:121]:
```suggestion
> Table 1. Atomic controls. Some advanced options are omitted.
```

File: docs/src/controlling_iterative_models.md [hunk 11]
```diff
@@ -223,14 +223,14 @@ Under the hood, control in MLJIteration is implemented using
 [IterationControl.jl](https://github.com/ablaom/IterationControl.jl). Rather
 than iterating a training machine directly, we iterate a wrapped
 version of this object, which includes other information that controls
```
Comment (ablaom) [inline docs/src/controlling_iterative_models.md:225]:
```suggestion
version of this object, which includes other information that a control
```

File: docs/src/evaluating_model_performance.md [hunk 12]
```diff
@@ -6,8 +6,8 @@ For more on available performance measures, see
 [Performance Measures](performance_measures.md).
 
 In addition to hold-out and cross-validation, the user can specify
-their own list of train/test pairs of row indices for resampling, or
-define their own re-usable resampling strategies.
+their list of train/test pairs of row indices for resampling, or
+define their re-usable resampling strategies.
```
Comment (ablaom) [inline docs/src/evaluating_model_performance.md:10]:
```suggestion
an explicit list of train/test pairs of row indices for resampling, or
define new resampling strategies.
```

File: docs/src/evaluating_model_performance.md [hunk 13]
```diff
@@ -71,7 +71,7 @@ evaluate!(mach,
 
 ## User-specified train/test sets
 
-Users can either provide their own list of train/test pairs of row indices for resampling, as in this example:
+Users can either provide their list of train/test pairs of row indices for resampling, as in this example:
```
Comment (ablaom) [inline docs/src/evaluating_model_performance.md:74]:
```suggestion
Users can either provide an explicit list of train/test pairs of row indices for resampling, as in this example:
```

File: docs/src/evaluating_model_performance.md [hunk 14]
```diff
@@ -105,7 +105,7 @@ MLJBase.TimeSeriesCV
 
 ## Custom resampling strategies
 
-To define your own resampling strategy, make relevant parameters of
+To define your resampling strategy, make relevant parameters of
```
Comment (ablaom) [inline docs/src/evaluating_model_performance.md:108]:
```suggestion
To define a new resampling strategy, make relevant parameters of
```

File: docs/src/frequently_asked_questions.md [hunk 15]
```diff
@@ -11,7 +11,7 @@ API. Meta-algorithms (systematic tuning, pipelining, etc) remain
 python wrapped code, however.
 
 While ScikitLearn.jl provides the Julia user with access to a mature
-and large library of machine learning models, the scikit-learn API on
+and a large library of machine learning models, the scikit-learn API on
```
Comment (ablaom) [inline docs/src/frequently_asked_questions.md:14]:
```suggestion
and large library of machine learning models, the scikit-learn API on
```

File: docs/src/getting_started.md [hunk 16]
```diff
@@ -46,7 +46,7 @@ iris = DataFrames.DataFrame(iris);
 nothing # hide
 ```
 
-Next, let's split the data "horizontally" into input and target parts,
+Next, let's split the data "horizontally" into the input and the target parts,
```
Comment (ablaom) [inline docs/src/getting_started.md:49]:
```suggestion
Next, let's split the data "horizontally" into input and target parts,
```

File: docs/src/glossary.md [hunk 17]
```diff
@@ -4,38 +4,38 @@ Note: This glossary includes some detail intended mainly for MLJ developers.
...
-### fit-result (type generally defined outside of MLJ)
+### fitresult (type generally defined outside of MLJ)
 
 Also known as "learned" or "fitted" parameters, these are "weights",
-"coefficients", or similar paramaters learned by an algorithm, after
+"coefficients", or similar parameters learned by an algorithm, after
 adopting the prescribed hyper-parameters. For example, decision trees
 of a random forest, the coefficients and intercept of a linear model,
-or the rotation and projection matrices of PCA reduction scheme.
+or the rotation and projection matrices of the PCA reduction scheme.
```
Comment (ablaom) [inline docs/src/glossary.md:32]:
```suggestion
or the projection matrices of a PCA dimension reduction algorithm.
```

File: docs/src/glossary.md [hunk 18]
```diff
@@ -4,38 +4,38 @@ Note: This glossary includes some detail intended mainly for MLJ developers.
...
 
 
 ### operation
 
 Data-manipulating operations (methods) parameterized by some
```
Comment (ablaom) [inline docs/src/glossary.md:37]:
```suggestion
Data-manipulating operations (methods) dispatched on some
```

File: docs/src/glossary.md [hunk 19]
```diff
@@ -4,38 +4,38 @@ Note: This glossary includes some detail intended mainly for MLJ developers.
...
-fit-result. For supervised learners, the `predict`, `predict_mean`,
+fitresult. For supervised learners, the `predict`, `predict_mean`,
```
Comment (ablaom) [inline docs/src/glossary.md:38]:
```suggestion
`fitresult`. For supervised learners, the `predict`, `predict_mean`,
```

File: docs/src/known_issues.md [hunk 20]
```diff
@@ -2,7 +2,7 @@
 
 Routine issues are posted
 [here](https://github.com/alan-turing-institute/MLJ.jl/issues). Below
-are some longer term issues and limitations.
+are some longer-term issues and limitations.
```
Comment (ablaom) [inline docs/src/known_issues.md:5]: "longer" is a bone fide  adjective, not a noun acting as an adjective.
```suggestion
are some longer term issues and limitations.
```

File: docs/src/learning_curves.md [hunk 21]
```diff
@@ -37,7 +37,7 @@ plot(curve.parameter_values,
 
 ![](img/learning_curve42.png)
 
-In the case the `range` hyperparameter is the number of iterations in
+In case the `range` hyperparameter is the number of iterations in
```
Comment (ablaom) [inline docs/src/learning_curves.md:40]:
```suggestion
If the `range` hyperparameter is the number of iterations in
```

File: docs/src/linear_pipelines.md [hunk 22]
```diff
@@ -25,7 +25,7 @@ We would like to train using a K-nearest neighbor model, but the
 model type `KNNRegressor` assumes the features are all
 `Continuous`. This can be fixed by first:
 
-- coercing the `:age` feature to have `Continuous` type by replacing
+- coercing the `:age` feature to have the `Continuous` type by replacing
```
Comment (ablaom) [inline docs/src/linear_pipelines.md:28]:
```suggestion
- coercing the `:age` feature to have `Continuous` type by replacing
```

File: docs/src/machines.md [hunk 23]
```diff
@@ -117,7 +117,7 @@ feature_importances(mach::Machine) -> vector_of_pairs
 Here a `vector_of_pairs` is a vector of elements of the form `feature => importance_value`,
 where `feature` is a symbol. For example, `vector_of_pairs = [:gender => 0.23, :height =>
 0.7, :weight => 0.1]`. If a model does not support feature importances for some model
-hyper-parameters, every `importance_value` will be zero. This kind of accesss is supported
+hyper-parameters, every `importance_value` will be zero. This kind of access is supported
```
Comment (ablaom) [inline docs/src/machines.md:120]: For consistency with earlier corrections.

```suggestion
hyperparameters, every `importance_value` will be zero. This kind of access is supported
```

File: docs/src/performance_measures.md [hunk 24]
```diff
@@ -88,7 +88,7 @@ measures(conditions...)
 ```
 
 A user-defined measure in MLJ can be passed to the `evaluate!`
-method, and elsewhere in MLJ, provided it is a function or callable
+method, and elsewhere in MLJ, provided it is a function or a callable
```
Comment (ablaom) [inline docs/src/performance_measures.md:91]:
```suggestion
method, and elsewhere in MLJ, provided it is a function or callable
```

File: docs/src/quick_start_guide_to_adding_models.md [hunk 25]
```diff
@@ -6,16 +6,16 @@ condensed instructions below, we assume: (i) you have a Julia
...
 these models with MLJ; and (iii) that you have a rough understanding
-of how things work with MLJ.  In particular you are familiar with:
+of how things work with MLJ.  In particular, you are familiar with:
 
 - what [scientific types](https://github.com/JuliaAI/ScientificTypes.jl) are
 
 - what `Probabilistic`, `Deterministic` and `Unsupervised` models are
 
-- the fact that MLJ generally works with tables rather than bare bone
+- the fact that MLJ generally works with tables rather than barebone
```
Comment (ablaom) [inline docs/src/quick_start_guide_to_adding_models.md:15]:
```suggestion
- the fact that MLJ generally works with tables rather than
```

File: docs/src/quick_start_guide_to_adding_models.md [hunk 26]
```diff
@@ -96,7 +96,7 @@ That macro specifies:
 * Constraints on the hyperparameters where `_` refers to a value passed.
 
 Further to the last point, `a::Float64 = 0.5::(_ > 0)` indicates that
-the field `a` is a `Float64`, takes `0.5` as default value, and
+the field `a` is a `Float64`, takes `0.5` as a default value, and
```
Comment (ablaom) [inline docs/src/quick_start_guide_to_adding_models.md:99]:
```suggestion
the field `a` is a `Float64`, takes `0.5` as its default value, and
```

File: docs/src/transformers.md [hunk 27]
```diff
@@ -78,7 +78,7 @@ transform(mach, [1, 2, 3], [3, 2, 1])
 
 Let's see how we can include our `Averager` in a learning network (see
 [Composing Models](@ref)) to mix the predictions of two regressors,
-with one-hot encoding of the inputs:
+with the one-hot encoding of the inputs:
```
Comment (ablaom) [inline docs/src/transformers.md:81]:
```suggestion
with one-hot encoding of the inputs:
```

**Review by ablaom — CHANGES_REQUESTED** [review]
Wow. Despite the add of grammarly, I can see real intelligence and attention to detail in this overdue review of the documentation. Many thanks 🙏🏾

File: docs/src/adding_models_for_general_use.md (hunk 6 above)
Comment (svilupp) [reply docs/src/adding_models_for_general_use.md:803]: Sure thing. I've double-checked that on Google and Wiki and that was also the more common use but happy to revert it.

**Comment by svilupp** [comment]
Sorry, you had to do so many reversals!

File: docs/src/known_issues.md (hunk 20 above)
Comment (svilupp) [reply docs/src/known_issues.md:5]: I went with the form used by Oxford Dictionary and Wikipedia but both are used.

**Review by ablaom — APPROVED** [review]
Assuming tests pass, this is good to go.

Thanks again @svilupp for your contribution 🙏🏾
